src/api/uploads/MultiputUpload.js

```diff
@@ -0,0 +1,358 @@
+/**
+ * @flow
+ * @file Multiput upload
+ * @author Box
+ */
+
+import Base from '../Base';
+import { getFileLastModifiedAsISONoMSIfPossible } from '../../util/uploads';
+import createWorker from '../../util/uploadsSHA1Worker';
+import MultiputPart, { PART_STATE_UPLOADED, PART_STATE_DIGEST_READY } from './MultiputPart';
+import type { StringAnyMap, MultiputConfig } from '../../flowTypes';
+
+const DEFAULT_MULTIPUT_CONFIG: MultiputConfig = {
+    console: false, // Whether to display informational messages to console
+    digestReadahead: 5, // How many parts past those currently uploading to precompute digest for
+    initialRetryDelayMs: 5000, // Base for exponential backoff on retries
+    maxRetryDelayMs: 60000, // Upper bound for time between retries
+    parallelism: 5, // Maximum number of parts to upload at a time
+    requestTimeout: 120000, // Idle timeout on part upload, overall request timeout on other requests
+    // eslint-disable-next-line max-len
+    retries: 5 // How many times to retry requests such as upload part or commit. Note that total number of attempts will be retries + 1 in worst case where all attempts fail.
+};
+
+class MultiputUpload extends Base {
+    clientId: ?string;
+    commitRetryCount: number;
+    config: Object;
+    createSessionNumRetriesPerformed: number;
+    destinationFile: ?string;
+    destinationFolder: ?string;
+    file: File;
+    fileSha1: ?string;
+    firstUnuploadedPartIndex: number;
+    initialFileLastModified: ?string;
+    initialFileSize: number;
+    onCompleted: ?Function;
+    onFailure: ?Function;
+    onProgress: ?Function;
+    options: Object;
+    partSize: number;
+    parts: Array<MultiputPart>;
+    partsDigestComputing: number;
+    partsDigestReady: number;
+    partsNotStarted: number;
+    partsUploaded: number;
+    partsUploading: number;
+    sessionEndpoints: Object;
+    sessionId: ?string;
+    totalUploadedBytes: number;
+    worker: Worker;
+
+    /**
+     * [constructor]
+     * 
+     * @param {object} options
+     * @param {File} file
+     * @param {string} createSessionUrl
+     * @param {string} [destinationFolder] - Untyped folder id (e.g. no "d_" prefix)
+     * @param {string} [destinationFile] - Untyped file id (e.g. no "f_" prefix)
```

Comment:
Clearer to use Id here - destinationFolderId, destinationFileId

Reply:
done